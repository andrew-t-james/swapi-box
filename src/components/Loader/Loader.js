import React, { Component } from 'react';
import { PulseLoader } from 'halogenium';

class Loader extends Component {
  render() {
    return (
      <div className="loader">
        <PulseLoader color="#F9D749" size="16px" margin="4px" />
      </div>
    );
  }
}

export default Loader;