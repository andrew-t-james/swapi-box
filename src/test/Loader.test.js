import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../components/Loader/Loader';

describe('Loader', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(
    <Loader />
  ));

  afterEach(() => wrapper.unmount());

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
