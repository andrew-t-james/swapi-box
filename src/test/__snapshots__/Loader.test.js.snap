// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <PulseLoader
        color="#F9D749"
        loading={true}
        margin="4px"
        size="16px"
/>,
      "className": "loader",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "#F9D749",
        "loading": true,
        "margin": "4px",
        "size": "16px",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <PulseLoader
          color="#F9D749"
          loading={true}
          margin="4px"
          size="16px"
/>,
        "className": "loader",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#F9D749",
          "loading": true,
          "margin": "4px",
          "size": "16px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
